api/src/main/java/org/pmiops/workbench/cdr/CdrDbConfig.java

```diff
@@ -81,13 +79,33 @@ public CdrDataSource(CdrVersionDao cdrVersionDao,
         String dbName = isWorkbenchDbUser ? cdrVersion.getCdrDbName() : cdrVersion.getPublicDbName();
         int slashIndex = originalDbUrl.lastIndexOf('/');
         String dbUrl = originalDbUrl.substring(0, slashIndex + 1) + dbName;
-        DataSource dataSource = DataSourceBuilder
-            .create()
-            .driverClassName(dbDriverClassName)
+
+        DataSource dataSource =
+            DataSourceBuilder.create()
+            .driverClassName(poolConfiguration.getDriverClassName())
             .username(dbUser)
             .password(dbPassword)
             .url(dbUrl)
             .build();
+        if (dataSource instanceof org.apache.tomcat.jdbc.pool.DataSource) {
```

Comment:
When is this not a Tomcat data source? (Tests?)

Reply:
Yeah, just tests (or if we changed the config). I added a warning log here to be safe.